supplementary/alica_ros1/alica_tests/src/test/test_alica_condition_plantype.cpp

```diff
@@ -53,43 +59,21 @@ TEST_F(AlicaConditionPlanType, conditionPlanTypeTest)
     ae->start();
     auto* wm = dynamic_cast<alicaTests::TestWorldModel*>(ac->getWorldModel());
 
-    for (int i = 0; i < 21; i++) {
-        ac->stepEngine();
-
-        //		if(i > 1)
-        //		{
-        //			long id =
-        //(ae->getPlanBase().getRootNode()->getChildren()[0]->getActiveState()->getId();
-        //			string name =
-        //(ae->getPlanBase().getRootNode()->getChildren()[0]->getActiveState()->getName();
-        //			cout << name << " : " << id  << " Iteration : " << i << endl;
-        //		}
-        if (i == 2) {
-            // Should be OtherPlan --> State
-            EXPECT_TRUE(alica::test::Util::isStateActive(ae, 1418042819204));
-        }
-        if (i == 5) {
-            wm->setRuntimeCondition1418042967134(true);
-        }
-        if (i == 6) {
-            // Should be RunTimeCondition --> State
-            EXPECT_TRUE(alica::test::Util::isStateActive(ae, 1418042806576));
-        }
-        if (i == 10) {
-            wm->setRuntimeCondition1418042967134(false);
-        }
-        if (i == 12) {
-            // Should be OtherPlan --> State
-            EXPECT_TRUE(alica::test::Util::isStateActive(ae, 1418042819204));
-        }
-        if (i == 13) {
-            wm->setPreCondition1418042929966(true);
-        }
-        if (i > 14) {
-            // Should be PreCondition --> State
-            EXPECT_TRUE(alica::test::Util::isStateActive(ae, 1418042796752));
-        }
-    }
+    CounterClass::called = 0;
+    STEP_UNTIL(alica::test::Util::isStateActive(ae, 1418042819204));
+    EXPECT_TRUE(alica::test::Util::isStateActive(ae, 1418042819204));
+    STEP_UNTIL(CounterClass::called == 5);
```

Comment:
why do we need to step 5 times for setting the runtime condition? Did you check what the test is trying to do?